provider/cluster/inventory.go

```diff
@@ -490,15 +555,17 @@ loop:
 			}
 
 			select {
-			case _ = <-is.readych:
+			case <-is.readych:
 				break
 			default:
 				is.log.Debug("inventory ready")
 				close(is.readych)
 			}
 
-			inventory = res.Value().(ctypes.Inventory)
-			metrics := inventory.Metrics()
+			resultArray := res.Value().([]interface{})
+
+			state.inventory = resultArray[0].(ctypes.Inventory)
```

Comment:
wen generics

Reply:
I feel like it should check `len(resultsArray)` and also use the two-lhs form of casting.  Any reason to not?

Reply:
I can double check this, I think there was some reason I wound up casting this here.

Reply:
I can just add some sort of local type here that gets returned as the result I think